### PR TITLE
HHH-11023 - Startup performance regression : schema update/validate

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/mapping/Index.java [hunk 1]
```diff
@@ -223,7 +224,22 @@ public String getName() {
 	}
 
 	public void setName(String name) {
-		this.name = name;
+		if (
+				StringHelper.isNotEmpty( name ) &&
+						Dialect.QUOTE.indexOf( name.charAt( 0 ) ) > -1 //TODO: deprecated, remove eventually
```
Comment (vladmihalcea) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: Maybe we should use Identifier.isQuoted here

File: hibernate-core/src/main/java/org/hibernate/mapping/Index.java [hunk 2]
```diff
@@ -223,7 +224,22 @@ public String getName() {
...
+				) {
+			quoted = true;
+			this.name = name.substring( 1, name.length() - 1 );
```
Comment (vladmihalcea) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: I think we might want to add an Identifier#unquote method and use it here as well

File: hibernate-core/src/main/java/org/hibernate/mapping/Index.java [hunk 3]
```diff
@@ -223,7 +224,22 @@ public String getName() {
...
+		}
+		else {
+			this.name = name;
+		}
+	}
+
+	public String getQuotedName(Dialect d) {
+		return quoted ?
+				d.openQuote() + name + d.closeQuote() :
```
Comment (vladmihalcea) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: Same as above. Can't we use Identifier#quote here?

File: hibernate-core/src/main/java/org/hibernate/mapping/Index.java (hunk 1 above)
Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: yes, it's better

File: hibernate-core/src/main/java/org/hibernate/mapping/Index.java (hunk 3 above)
Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: probably we have to change the Index property name from String to Identifier

File: hibernate-core/src/main/java/org/hibernate/mapping/Index.java (hunk 2 above)
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: There is `org.hibernate.internal.util.StringHelper#unquote`, as well as `Identifier.toIdentifier( name ).getText()`

File: hibernate-core/src/main/java/org/hibernate/mapping/Index.java (hunk 3 above)
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: Yes that is a general change we need to make all across `org.hibernate.mapping`.  We will focus on that in 7.  But we can certainly change little things like this piecemeal.

File: hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java [hunk 4]
```diff
@@ -1311,6 +1311,20 @@
...
+	String HBM2DDL_SCHEMA_MIGRATOR = "hibernate.hbm2ddl.schema_migrator";
+
+	/**
+	 * Used to specify the {@link org.hibernate.tool.schema.spi.SchemaValidator} to be used to
+	 * validate a database schema.
+	 *
+	 */
+	String HBM2DDL_SCHEMA_VALIDATOR = "hibernate.hbm2ddl.schema_validator";
+
+	/**
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java]: If you are adding the ability to swap the validator and migrator delegates, why not the dropper and creator as well?

File: hibernate-core/src/main/java/org/hibernate/mapping/Index.java (hunk 1 above)
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: As indicated below (as well as Christian's PR) really the proper solution imo is to track the Identifier here, not the String.

File: hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java (hunk 4 above)
Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java]: @sebersole 
2 reasons, first I have not thought about it :) and second not sure adding the ability to swap the validator and migrator delegates is the best solution, may be as you suggested me a boolean config setting may be better.
Not sure about the config property name any idea?

**Comment by dreab8** [comment]
https://github.com/hibernate/hibernate-orm/pull/1574 replaces this PR

**Pull request closed without merging**